go/chat/attachment_httpsrv.go

```diff
@@ -521,9 +525,27 @@ func (r *AttachmentHTTPSrv) serveAttachment(ctx context.Context, w http.Response
 		}
 		http.ServeContent(w, req, asset.Filename, time.Time{}, rs)
 	} else {
-		if err := r.fetcher.FetchAttachment(ctx, w, pair.ConvID, asset, r.ri, r, blankProgress); err != nil {
-			r.makeError(ctx, w, http.StatusInternalServerError, "failed to fetch attachment: %s", err)
-			return
+		// no animation mode is intended to transform GIF images into single frame versions
+		if noAnim {
+			var buf bytes.Buffer
+			if err := r.fetcher.FetchAttachment(ctx, &buf, pair.ConvID, asset, r.ri, r, blankProgress); err != nil {
+				r.makeError(ctx, w, http.StatusInternalServerError, "failed to fetch attachment: %s", err)
+				return
+			}
+			bufReader := attachments.NewBufReadResetter(buf.Bytes())
+			if err := attachments.GIFToPNG(ctx, bufReader, w); err != nil {
```

Comment:
should we cache these?

Reply:
Should be basically instant, don't think we need to.

Reply:
I guess a small LRU couldn't hurt though